tests/layers/test_numerical_calibrator.py

```diff
@@ -170,6 +199,32 @@ def test_forward(input_keypoints, kernel_init, kernel_data, inputs, expected_out
     assert torch.allclose(outputs, expected_outputs)
 
 
+@pytest.mark.parametrize(
+    "input_keypoints",
+    [
+        (np.linspace(1, 4, num=4)),
+        (np.array([0.0, 1.2, 2.0, 3.7, 5.0])),
+        (np.linspace(1, 20, num=45)),
+        (np.array([0.0, 0.02, 0.023, 3.7, 5.0, 7.9, 9.9, 10.3, 12.4, 15.6, 51.2])),
+    ],
+)
+@pytest.mark.parametrize(
+    "x",
+    [(torch.tensor([[1.1]])), (torch.tensor([[1.2], [1.3]]))],
+)
+def test_forward_learned_inputs(input_keypoints, x):
+    """Tests that interior keypoints are properly reconstructed for learned inputs."""
```

Comment:
update docstring here as well wrt. learned inputs -> learned input keypoints. If it's too long you might need to rework the sentence.